OPHD/Things/Structures/MineFacility.h

```diff
@@ -23,6 +23,10 @@ class MineFacility: public Structure
 
 	int digTimeRemaining() const;
 
+	int assignedTrucks()  const { return mAssignedTrucks; }
+	void addTruck() { mAssignedTrucks = std::clamp(++mAssignedTrucks, 1, 10); }
+	void removeTruck() { mAssignedTrucks = std::clamp(--mAssignedTrucks, 1, 10); }
```

Comment:
Based on context here, it seems you only want to modify the variable when it falls within a specified bounds. As such, I would avoid using `++mAssignedTrucks` and `--mAssignedTrucks`, which would unconditionally modify the variables. Instead, pass `mAssignedTrucks + 1`, and `mAssignedTrucks - 1`.

Reply:
Yup, that's precisely what I was going for. This is why so-called 'clever' programming is almost always a bad idea. While the code behaved as expected on the MSVC++ compiler, it can be ambiguous and could have unintended side effects.

In this case I'd rather use the much more clear `variable + 1` option.